vignettes/design_principle.Rmd

```diff
@@ -0,0 +1,196 @@
+
+Our first attempt consisted in using the currently available R packages designed to access data from specific HIS. These packages are usually tied to a single HIS and can't be used to query others. {fingertipsR}, {REDCapR}, {godataR}, {globaldothealth} can be used to fetch data from [Fingertips](https://fingertips.phe.org.uk/), [REDCap](https://projectredcap.org/software/), [goData](https://www.who.int/tools/godata), and [Global.Health](https://global.health/) respectively. In version **0.0.1** of the {readepi} package, when a user is requesting data from a specific HIS, its correspondent package is called internally. 
+
+As each packages was designed to target a specific HIS, this approach increases our dependency to many other packages and introduces the challenge of having a unified framework for importing data from multiple HIS.
+
+To address this challenge, PIs of the [Epiverse-TRACE](https://data.org/initiatives/epiverse/) came with a list of potential data sources for which we aim at building a tool to request and fetch the data of interest from multiple source in the same way. The data sources include: distributed health information systems, and public databases as shown in the figure below.
+
+![readepi roadmap](../man/figures/roadmap_readepi.drawio.svg)
+
+## Scope
+
+The {readepi} package intends to import data from two common sources of institutional health-related data: health information systems (HIS) wrapped with specific Application programming interfaces (APIs) and relational database management systems (RDBMS) that run on specific servers.
+
+Importing data from any of these sources requires the user to have the right access. The user is also expected to provide the relevant query parameters to fetch the target data.
+Hence, the {readepi} package is structured around one main function (`read()`) and two auxiliary functions (`authenticate()` and `get_metadata()`).
+
+The previous version of {readepi} (`0.0.1`) supports importing data from HIS APIs such as REDCap (Research Electronic Data Capture), DHIS2 (District Health Information System 2), and Fingertips as well as RDBMS such as MS SQL, SQLite, MYSQL, and PostgreSQL. 
+
+In the next versions, the `read()` function will also allow data import from HIS like GoData, Globaldothealth, and SORMAS. It will also include functionalities for importing data from RDBMS such as MS ACCESS, and SQlite.
+
+<!-- ![readepi design diagram](../man/figures/design_vignette_diagram.drawio.svg) -->
+
+
+## Output
+
+The `read()` functions return a `list` object containing one or more `data frames`. Each `data frame` corresponds to the data from a specified source.
+The `get_metadata()` function returns a data dictionary containing information about the data structure.
+The `authenticate()` function returns a connection object that is used in the query request.
```

Comment:
Am I misunderstanding something or does this not match what the diagram is describing (`read_from_api()`, `read_from_server()`, `authenticate()`)?

Reply:
There are few changes in the design diagram. We now want to split `read_from_api()` into small functions that are specific to a given HIS. This, we think, will make it easy to perform targetted updates on the way data is imported from a HIS, when there is any changes from that API.

I have updated the design diagram in commit `41f41e8`.